user_management/api/serializers.py

```diff
@@ -8,18 +8,24 @@
 User = get_user_model()
 
 
-class ValidateEmailMixin(object):
-    def validate_email(self, attrs, source):
-        email = attrs.get(source).lower()
+class UniqueEmailValidator(validators.UniqueValidator):
+    def filter_queryset(self, value, queryset):
+        """Check lower-cased email is unique."""
+        return super(UniqueEmailValidator, self).filter_queryset(
+            value.lower(),
+            queryset,
+        )
 
-        try:
-            User.objects.get_by_natural_key(email)
```

Comment:
Do you think we should continue to support that use-case directly here or should such projects handle it themselves? If we should support it here, can you add or sketch a test for this?